### PR TITLE
Improve Switch to Expensify Classic flow

### PR DESCRIPTION
**Pull request by hungvu193**
### Explanation of Change

Update the Switch to Expensify Classic survey to drive better feedback
### Fixed Issues

$ https://github.com/Expensify/App/issues/51703
PROPOSAL: N/A

### Tests

**Case 1: People that are redirected into NewDot**
1. Go to [expensify.com](http://expensify.com/)
2. Choose the “1-9” signUpQualifier
3. Sign-up for a new account
4. Verify you’re redirected to [new.expensify.com](http://new.expensify.com/)
5. Click the Settings bottom tab
6. Click `Switch to Expensify Classic`
7. Verify you see the new “Would you like to speak to a product manager?” page.
8. Click “Book a call”
9. Verify a Calendly link opens in a new browser tab
10. Go back to the [new.expensify.com](http://new.expensify.com/) tab
11. Verify the RHP modal has closed.
12. Click `Switch to Expensify Classic`
13. Click `No thanks`
14. Choose any of the three radio button options
15. Click `Next`
16. Don’t enter any text in the free form text box
17. Click `Next`
18. Verify there’s no error on the page
19. Click `Switch to Expensify Classic`
20. Verify you’re redirected to [expensify.com](http://expensify.com/)

**Case 2: People not redirected into NewDot**
1. Go to [expensify.com](http://expensify.com/)
2. Choose the “10+” signUpQualifier
3. Sign-up for a new account
4. Verify you land in [expensify.com](http://expensify.com/)
5. Click the Support button in the LHN of OldDot
6. Click Concierge
7. Verify you’re redirected to [new.expensify.com](http://new.expensify.com/)
8. Click the Settings bottom tab
9. Click `Switch to Expensify Classic`
10. Verify you see the “Quick tip” page only
11. Click “Take me to Expensify Classic”
12. Verify you’re redirected to [expensify.com](http://expensify.com/)
- [x] Verify that no errors appear in the JS console

### Offline tests

N/A
### QA Steps

Same with Tests.

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/b93eecf3-785a-4d88-949f-b131b178db4b

https://github.com/user-attachments/assets/d31d1ef1-13f7-490a-a9b3-cc2f1a7e4f35

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/daafbdbf-f178-4a00-a702-8854c1ac87b6

https://github.com/user-attachments/assets/1eb3bd93-5055-47a4-9edd-88cdf2aaed24

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/18f0ce6b-d859-4188-8bc6-78f9ecb69a78

https://github.com/user-attachments/assets/86dc35da-8be7-4bd4-a485-67293888f1c9

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/89bcfe46-575f-4b20-939d-fa9abebb4287

https://github.com/user-attachments/assets/0108b77e-667e-4d0d-be09-da09f1b3aca1

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/b68d2175-feca-40b5-b712-aaddb401f952

https://github.com/user-attachments/assets/2f74f0a6-0750-4c0d-9d3e-9b1d19165be4
</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/2f74f0a6-0750-4c0d-9d3e-9b1d19165be4

https://github.com/user-attachments/assets/b68d2175-feca-40b5-b712-aaddb401f952

</details>

### CODE REVIEW

**Comment by hungvu193** [comment]
**Case 1:**

https://github.com/user-attachments/assets/2f74f0a6-0750-4c0d-9d3e-9b1d19165be4

**Case 2:**

https://github.com/user-attachments/assets/b68d2175-feca-40b5-b712-aaddb401f952

File: src/languages/es.ts [hunk 1]
```diff
@@ -5442,6 +5442,18 @@ const translations = {
...
+        bookACall: 'Reservar una llamada',
+        noThanks: 'No, gracias',
+        bookACallTitle: '¿Desea hablar con un responsable de producto?',
+        benefits: {
+            [CONST.EXIT_SURVEY.BENEFIT.CHATTING_DIRECTLY]: 'Charla directa sobre gastos e informes',
+            [CONST.EXIT_SURVEY.BENEFIT.EVERYTHING_MOBILE]: 'Posibilidad de hacerlo todo desde el móvil',
+            [CONST.EXIT_SURVEY.BENEFIT.TRAVEL_EXPENSE]: 'Viajes y gastos a la velocidad del chat',
+        },
+        bookACallTextTop: 'Al cambiar a Expensify Classic, se perderá:',
+        bookACallTextBottom: 'Nos encantaría hablar con usted para entender por qué. Puede concertar una llamada con uno de nuestros jefes de producto para hablar de sus necesidades.',
```
[the remark this answers is not in the document]
Comment (trjExpensify) [reply src/languages/es.ts:5456]: Asked @Gonals to take a look to keep this moving!

**Comment by trjExpensify** [comment]
Buildin'...

**Comment by trjExpensify** [comment]
@hungvu193 can we make sure the button label on the quickTip page reads `Take me to Expensify Classic`? 

We purposely had a slightly different label name for that, so we can set up tracking it separately in FullStory.

**Comment by hungvu193** [comment]
> @hungvu193 can we make sure the button label on the quickTip page reads `Take me to Expensify Classic`? 
> 
> 
> 
> We purposely had a slightly different label name for that, so we can set up tracking it separately in FullStory. 

Sure. I think I missed that case, will update it along with the new translations.

File: src/languages/es.ts [hunk 2]
```diff
@@ -5442,6 +5442,18 @@ const translations = {
         offlineTitle: 'Parece que estás atrapado aquí...',
         offline:
             'Parece que estás desconectado. Desafortunadamente, Expensify Classic no funciona sin conexión, pero New Expensify sí. Si prefieres utilizar Expensify Classic, inténtalo de nuevo cuando tengas conexión a internet.',
+        quickTip: 'Consejo rápido...',
+        quickTipSubTitle: 'Puedes ir directamente a Expensify Classic visitando expensify.com. Márcalo como favorito para tener un acceso directo fácil.',
+        bookACall: 'Reservar una llamada',
+        noThanks: 'No, gracias',
+        bookACallTitle: '¿Desea hablar con un responsable de producto?',
+        benefits: {
+            [CONST.EXIT_SURVEY.BENEFIT.CHATTING_DIRECTLY]: 'Charla directa sobre gastos e informes',
```
Comment (trjExpensify) [inline src/languages/es.ts:5451]:
```suggestion
            [CONST.EXIT_SURVEY.BENEFIT.CHATTING_DIRECTLY]: 'Chat directo sobre gastos e informes',
```

As per: https://expensify.slack.com/archives/C01GTK53T8Q/p1731587710343179?thread_ts=1731572881.369449&cid=C01GTK53T8Q

**Comment by trjExpensify** [comment]
Great, I made the suggestion on the one translation change.

@shubham1206agra is this a PR you'll be able to review today? If not, please do let me know so I can find a replacement. :)

File: src/languages/es.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (trjExpensify) [reply src/languages/es.ts:5456]: Translation for "Take me to Expensify Classic" === `Llévame a Expensify Classic`

**Comment by trjExpensify** [comment]
Alright, did some testing. Overall looking good! 

- [tom+1to9switch@trj.chat](mailto:tom+1to9switch@trj.chat) -  Signed-up on expensify.com, selected 1-9 to be redirected to NewDot and get `tryNewDot.classicRedirect.dismissed===false` set for the account. I was shown the survey flow, book a call looks good etc. Then I switched, such that `classicRedirect.dismissed` becomes `true`. That looks good as well! Only thing to fix there is the label name on the quickTip page, as discussed above.

https://github.com/user-attachments/assets/982068d6-14a6-4015-ac21-30c94c5c9bdb

- [tom+10plusswitch@trj.chat](mailto:tom+10plusswitch@trj.chat) - signed-up on expensify.com and selected the 10+ option this time, so I have no tryNewDot NVP set at all for this account. We show the quickTrip page. That looks good!

https://github.com/user-attachments/assets/db3fb6eb-5f17-4518-9518-7ab9546695dd


- tom+dksksk@trj.chat - I signed-up on NewDot directly with this account. The tryNewDot NVP is added to the account, but classicRedirect.dismissed is empty (not set to `false` or `true`). That put me in a weird place where I saw the last page of the survey flow only for some reason. 

https://github.com/user-attachments/assets/7072a6c7-42ac-4ab9-a602-1c2164f425a5

So two things on that bug: 

1. We should show the quickTip page if `tryNewDot` NVP exists with `classicRedirect.dismissed` empty. Let’s account for that here in this PR with our logic, in case it ever happens. 
2. I think we should be setting people who sign-up directly on NewDot to `tryNewDot.classicRedirect.dismissed===false`. That’s probably out of scope of this PR, but flagging it here as I spotted it. CC: @Julesssss for a second opinion on that.

**Comment by hungvu193** [comment]
`thrown: "Exceeded timeout of 30000 ms for a test.` I don't think it's related to this PR.

**Comment by hungvu193** [comment]
@trjExpensify Can you try to check the Onyx key `nvp_tryNewDot` in the bug that you posted? I'm unable to reproduce that bug so that will be helpful

Screenshot 2024-11-14 at 21 23 09

**Comment by trjExpensify** [comment]
> @trjExpensify Can you try to check the Onyx key `nvp_tryNewDot` in the bug that you posted? I'm unable to reproduce that bug so that will be helpful
> 
> Screenshot 2024-11-14 at 21 23 09


Yeah, it's empty: 
image

**Comment by trjExpensify** [comment]
I've figured it out. 

- If you sign-up with a public domain on NewDot, we set tryNewDot.classicRedirect.dismissed===false. 
- If you sign-up with a private domain on NewDot, we set tryNewDot NVP but classicRedirect.dismissed is empty.

**Comment by hungvu193** [comment]
Cool cool. We can fix this on FE anyway. I've just pushed the change 🚀

**Comment by trjExpensify** [comment]
It's good! 

image

**Review by mountiny — COMMENTED** [review]
Code looks good ot me, just waiting for the checklist

**Comment by trjExpensify** [comment]
> 
> 
> https://github.com/user-attachments/assets/63698a5d-a9f7-49c6-81a1-07ad76c5aa3a
> 
> 
> 
> BUG: Modal gets closed on clicking `Book a call` option.

This one is expected. We talked about it with design in the thread linked in the issue OP.

**Comment by hungvu193** [comment]
> Screen.Recording.2024-11-14.at.11.11.47.PM.mov 
> BUG: Page switches in background when clicked no thanks

This one also happens on staging

https://github.com/user-attachments/assets/c9523834-60f2-4500-9dad-209d717807b5

**Comment by hungvu193** [comment]
> Getting infinite load in NewDot prod on new account creation.

I don't think this is related to this PR 😄

**Comment by hungvu193** [comment]
Fixed the navigation issue. Please test again @shubham1206agra

**Comment by trjExpensify** [comment]
Alrighty, how we looking here?

**Review by shubham1206agra — APPROVED** [review]

**Review by mountiny — APPROVED** [review]
Thanks for the changes and testing

**Comment by mountiny** [comment]
@hungvu193 can you think of ways to add unit tests for these changes? Lets try to add it in a follow up. thanks!

**Comment by hungvu193** [comment]
> @hungvu193 can you think of ways to add unit tests for these changes? Lets try to add it in a follow up. thanks!

Sure. Lemme think about it.

**Comment by mvtglobally** [comment]
@mountiny This PR is not fully testable easily on Hybrid app. 
Anyway you can help us highlight what exactly can be verified on Hybrid since we dont see URLs to be definitive

**Pull request merged**